src/app/organization/holidays/edit-holiday/edit-holiday.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators, FormControl } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services. */
+import { OrganizationService } from 'app/organization/organization.service';
+
+/**
+ * Edit Holiday component.
+ */
+@Component({
+  selector: 'mifosx-edit-holiday',
+  templateUrl: './edit-holiday.component.html',
+  styleUrls: ['./edit-holiday.component.scss']
+})
+export class EditHolidayComponent implements OnInit {
+
+  /** Edit Holiday form. */
+  holidayForm: FormGroup;
+  /** Holiday data. */
+  holidayData: any;
+  /** Rescheduling Type. */
+  reSchedulingType: number;
+  /** Is Active Holiday. */
+  isActiveHoliday = true;
+  /** Minimum Date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum Date allowed. */
+  maxDate = new Date();
+
+  /**
+   * Get holiday and holiday template from `Resolver`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {DatePipe} datePipe Date Pipe.
+   * @param {OrganizationService} organizatioService Organization Service.
+   * @param {Router} router Router.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private datePipe: DatePipe,
+              private organizatioService: OrganizationService,
+              private router: Router ) {
+    this.route.data.subscribe((data: { holiday: any, holidayTemplate: any }) => {
+      this.holidayData = data.holiday;
+      this.holidayData.repaymentSchedulingTypes = data.holidayTemplate;
+      this.reSchedulingType = this.holidayData.reschedulingType;
+      if ( this.holidayData.status.value === 'Active' ) {
+        this.isActiveHoliday = true;
+      } else {
+        this.isActiveHoliday = false;
+      }
+    });
+  }
+
+  ngOnInit() {
+    this.setEditForm();
+    this.getReschedulingType();
+  }
+
+  /**
+   * Sets Edit Form.
+   */
+  setEditForm() {
+    this.holidayForm = this.formBuilder.group({
+      'name': [this.holidayData.name, Validators.required],
+      'description': [this.holidayData.description]
+    });
+
+    if (!this.isActiveHoliday) {
+      this.holidayForm.addControl('fromDate', new FormControl(this.holidayData.fromDate && new Date(this.holidayData.fromDate), Validators.required));
+      this.holidayForm.addControl('toDate', new FormControl(this.holidayData.toDate && new Date(this.holidayData.toDate), Validators.required));
+      this.holidayForm.addControl('reschedulingType', new FormControl(this.holidayData.reschedulingType, Validators.required));
+      if (this.reSchedulingType === 2) {
+        this.holidayForm.addControl('repaymentsRescheduledTo', new FormControl(this.holidayData.repaymentsRescheduledTo && new Date(this.holidayData.repaymentsRescheduledTo), Validators.required));
+      }
+    }
+  }
+
```

Comment:
check community app controller for logic

Reply:
I checked that. Are there any errors?

Reply:
Nope just making sure you did it.